src/Components/Server/src/Circuits/RemoteJSDataStream.cs

```diff
@@ -53,7 +41,7 @@ public static async Task<bool> ReceiveData(RemoteJSRuntime runtime, long streamI
                 Math.Min(maximumIncomingBytes, 50*1024) - 512 :
                 throw new ArgumentException($"SignalR MaximumIncomingBytes must be at least 1 kb.");
 
-            var streamId = runtime.RemoteJSDataStreamNextInstanceId++;
+            var streamId = runtime.JSDataStreamNextInstanceId++;
```

Comment:
Are all calls to this going thru the sync context (should we be concerned about concurrency at all here)?

Reply:
Yes, [we're going through the sync context](https://github.com/dotnet/aspnetcore/blob/8adbdc6d2559d55469aeaf1feae8061df5c27af2/src/Components/Server/src/Circuits/CircuitHost.cs#L431-L434) so parallel invocations of `ReceiveData` should not be a factor (hence `++` instead of `Increment`.